ReactAndroid/src/main/java/com/facebook/react/modules/location/LocationModule.java

```diff
@@ -188,18 +190,34 @@ public void stopObserving() {
     mWatchedProvider = null;
   }
 
+
+  private boolean isProviderEnabled(LocationManager locationManager, String provider) {
+    try {
+      // Before API version LOLLIPOP, this method would throw SecurityException
+      // if the location permissions were not sufficient to use the specified provider.
+      return locationManager.isProviderEnabled(provider);
+    } catch (SecurityException e) {
+      return false;
+    }
+  }
+
   @Nullable
-  private static String getValidProvider(LocationManager locationManager, boolean highAccuracy) {
+  private String getValidProvider(LocationManager locationManager, boolean highAccuracy) {
     String provider =
         highAccuracy ? LocationManager.GPS_PROVIDER : LocationManager.NETWORK_PROVIDER;
-    if (!locationManager.isProviderEnabled(provider)) {
+    if (!isProviderEnabled(locationManager, provider)) {
       provider = provider.equals(LocationManager.GPS_PROVIDER)
           ? LocationManager.NETWORK_PROVIDER
           : LocationManager.GPS_PROVIDER;
-      if (!locationManager.isProviderEnabled(provider)) {
+      if (!isProviderEnabled(locationManager, provider)) {
         return null;
       }
     }
+    // If it's an enabled provider, but we don't have permissions, ignore it
+    int finePermission = ContextCompat.checkSelfPermission(getReactApplicationContext(), android.Manifest.permission.ACCESS_FINE_LOCATION);
```

Comment:
What do you mean by:

> used COARSE but not FINE permissions

Does this mean declaring them in the manifest?

> "new FINE permission" that RN sometimes triggers accidentally

Why does RN accidentally need a new permission? That seems like a bug.

> If they are trying to use the GPS, but they don't have a FINE permission enabled, then pretend like we don't have a GPS

However, if we don't have the fine permission, we may still use Geolocation, correct?